ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/activity/PaywallActivityLauncher.kt

```diff
@@ -105,7 +109,7 @@ class PaywallActivityLauncher(resultCaller: ActivityResultCaller, resultHandler:
         val shouldDisplayBlock = shouldDisplayBlockForEntitlementIdentifier(requiredEntitlementIdentifier)
         shouldDisplayPaywall(shouldDisplayBlock) { shouldDisplay ->
             paywallDisplayCallback?.onPaywallDisplayResult(shouldDisplay)
-            if (shouldDisplay) {
+            if (shouldDisplay && !isActivityFinishing()) {
                 activityResultLauncher.launch(
```

Comment:
This is what would crash if the original activity is finished by the time it gets here.

Reply:
Would it be helpful to print a warning, or maybe even error, log to indicate why the Paywall didn't launch?

Reply:
Good suggestion! Will add something

Reply:
Added in https://github.com/RevenueCat/purchases-android/pull/1846/commits/20d9ad3b23d3dc736d6d288e5e76401bcb126fab